packages/@react-types/shared/src/events.d.ts

```diff
@@ -44,7 +44,11 @@ export interface PressEvent {
   /** Whether the meta keyboard modifier was held during the press event. */
   metaKey: boolean,
   /** Whether the alt keyboard modifier was held during the press event. */
-  altKey: boolean
+  altKey: boolean,
+  /** X position relative to the target. */
+  x: number,
```

Comment:
Thanks for the review. I agree that the coordinates should be `undefined` when triggered by the keyboard. 🤔

Another idea is to set the coordinates to the center of the target element when triggered by the keyboard. what do you think? 👀

Reply:
center of the element is definitely intriguing, though right now the position is clientX which is relative to the viewport, so we'd need to account for positioning in the center in the same way. That or maybe the position should be relative to the target and we should be using offsetX instead. What do you think?

Reply:
If you do a `getBoundingClientRect()` on an element, the x, y is the top left corner not the center. Would someone who wants the x, y of a keyboard event on something pressable want the corner or the center for whatever math they end up using it for? I do like the idea of the center more.

I'm trying to think about why I'd want an x, y from a keyboard event. I'd probably want the target. It feels like we should leave the x and y off the keyboard event like a native keyboard event.

Reply:
I'm leaning towards what @ktabors mentioned, do we have any use cases in mind that would require x,y coordinates for a keyboard event? We go with `undefined` coords for keyboard events and add something later if the need arises

Reply:
Ok, I had an offline conversation with the team, we'd like for coordinates to be relative to the target element. For any event that doesn't contain location information, such as keyboard, the center of the target element should be used for the x and y.

Which is what you've proposed here.

We don't see a reason to avoid doing the computation for keyboard to find the center. So lets keep x and y as required fields. I think for now we don't include the extra rect, it's one more thing to expose and I think the majority of cases are covered by the x, y we'll be providing. In addition, people can already get the target element, so they can compute more should they need to.